### DISCUSSION TITLE
How to reduce the size of exe packaging

### DISCUSSION THREAD

**Discussion by tsiens** [flet-dev/flet#5740] [Q&A]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

I used a conda virtual environment and pyinstaller to package my application, resulting in a size of 53.8MB. Using flet pack resulted in a package size of 77.8MB. When using flet build, the size went over 100MB... For version 2.5.2, installing only flet[desktop] and packaging with pyinstaller yielded a perfect 17MB. However, when running on other computers, it apparently requires downloading flet for Windows. The package created by flet pack automatically downloads and installs flet[cli], ending up being 77.8MB again.

I noticed there is a libmpv-2.dll file, sized 28.3MB, located in Lib\site-packages\flet_desktop\app\flet. It seems to provide MPV player functionality, which I don't use for audio/video features in my application. Can I remove this file?

excess，right click on the application on the Windows taskbar，flet's own name (flet.exe、flet descriptions) came up，I would like to display the application's own name and icon
pyinstall exe:
![image](https://github.com/user-attachments/assets/465404af-47b0-4b26-afa1-e4c5e133e0fb)
pack exe:
![image](https://github.com/user-attachments/assets/0044d88a-f9cb-4631-946d-ffcabc4e83a5)

### Code sample
```
import flet as ft


def main(page: ft.Page):
    page.window.icon = '../logo.ico'
    page.title = 'main'
    page.add(ft.Row([]))


if __name__ == '__main__':
    ft.app(target=main)
```
### To reproduce

```
flet pack main.py -y -n pack -i logo.ico
pyinstaller main.py -F -w -y -i logo.ico -n pyinstall --distpath dist --workpath build
```

### Operating System

Windows

### Operating system details

windows 11

### Flet version

2.5.2

### Regression

I'm not sure / I don't know

**Comment by josStorer** [comment]
The following information may be useful for those searching for related issues:

For me, I am currently still using v0.19.0, which I consider to be the best version in terms of the trade-off between size, startup speed, and functionality. When packaged with Nuitka and some unnecessary files removed, the typical size of the compressed package on windows is about 11MB, and there is no need to install any additional dependencies. It can start almost without any delay. All subsequent versions have significantly increased in size and become slow to start.

Regarding the startup speed issue, the latest version of Flet suggests using a startup screen to mitigate it. However, I believe it is still possible to restore the original fast startup (versions v0.19.0 and earlier), but considering that the project architecture of Flet has undergone significant changes since then, I think it would require an enormous amount of work to achieve this.

Related Issues: #3048 #2319 #3203 #5068 #5060 #3736 #4978

---

Here is a packaged example program [main.dist.zip](https://github.com/user-attachments/files/20368228/main.dist.zip) (8.65MB) that you can run in a sandbox environment. It's packed using python3.8 + nuitka2.7.3.

`nuitka .\main.py --windows-console-mode=disable --msvc=latest --standalone`.

Note that this packing method will download the flet-windows.zip runtime environment from GitHub during the first run to share the runtime environment with other programs and reduce the size of the program. You can proactively download the flet-windows.zip of v0.19.0 and package the environment together, which will increase the size of the compressed package to 17.7MB.
